src/matter/cluster/OnOffCluster.ts

```diff
@@ -4,18 +4,109 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { BooleanT } from "../../codec/TlvObjectCodec";
-import { Attribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { BooleanT, Field, ObjectT, StringT, Template, UnsignedIntT, UnsignedLongT } from "../../codec/TlvObjectCodec";
+import { Attribute, OptionalAttribute, OptionalWritableAttribute, Cluster, Command, NoArgumentsT, NoResponseT } from "./Cluster";
+import { TlvType } from "../../codec/TlvCodec";
 
+export const enum OnOffStartUpOnOff {
+    Off = 0,
+    On = 1,
+    Toggle = 2,
+}
+const OnOffStartUpOnOffT = { tlvType: TlvType.UnsignedInt } as Template<OnOffStartUpOnOff>;
+
+export const enum OnOffEffectIdentifier {
+    DelayedAllOff = 0,
+    DyingLight = 1,
+}
+const OnOffEffectIdentifierT = { tlvType: TlvType.UnsignedInt } as Template<OnOffEffectIdentifier>;
+
+export const enum OnOffDelayedAllOffEffectVariant {
+    FadeToOffIn_0p8Seconds = 0,
+    NoFade = 1,
+    "50PercentDimDownIn_0p8SecondsThenFadeToOffIn_12Seconds" = 2,
+}
+const OnOffDelayedAllOffEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDelayedAllOffEffectVariant>;
+
+export const enum OnOffDyingLightEffectVariant {
+    "20PercentDimUpIn_0p5SecondsThenFadeToOffIn_1Second" = 0,
+}
+const OnOffDyingLightEffectVariantT = { tlvType: TlvType.UnsignedInt } as Template<OnOffDyingLightEffectVariant>;
+
+/*
+  <bitmap name="OnOffControl" type="BITMAP8">
+    <cluster code="0x0006"/>
+    <field name="AcceptOnlyWhenOn" mask="0x01"/>
+  </bitmap>
+*/
+
+const OffWithEffectRequestT = ObjectT({
+    effectId: Field(0, OnOffEffectIdentifierT),
+    effectVariant: Field(1, OnOffDelayedAllOffEffectVariantT), /* OR OnOffDyingLightEffectVariantT depending on first field */
```

Comment:
ANy idea how to react for that? This is kind of a "combi field" and the allowed values and their meaning for effectVariant depends on effectId. Combine into one type and Do via "Naming" or whats best? ;-)

Reply:
>  Hopefully we won't see this one too often...

yes :-))

Reply:
Works ... tried something comparable but in wrong place :-)